### ISSUE TITLE
Option to treat float64 as float32 in write_vcf

### ISSUE THREAD

**Issue by timothymillar** [pystatgen/sgkit#1102]
The `write_vcf` function doesn't support float64 which makes sense given it's not supported by the VCF specification. However, manually converting float64 arrays to float32 is a bit of a pain, so I wonder if we could have a flag to treat float64 as float32? It's also a little inconsistent as we allow all integer types where the VCF spec states that the "Integer" type is int32 [(code)](https://github.com/pystatgen/sgkit/blob/main/sgkit/io/vcf/vcf_writer.py#L590).

**Comment by timothymillar** [comment]
Ah, I see that int64 fields result in errors elsewhere [(e.g.)](https://github.com/pystatgen/sgkit/blob/main/sgkit/io/vcf/vcf_writer_utils.py#L496)

**Comment by tomwhite** [comment]
I agree that it would be nice to support float64 arrays. The VCF Zarr spec covers them: https://github.com/pystatgen/vcf-zarr-spec/blob/main/vcf_zarr_spec.md, so it should be possible to update the mask method to support float64.
